motor_control/throttle/button_control.ino

```diff
@@ -1,29 +1,23 @@
-#include "button_control.h"
-
-ButtonThrottle::ButtonThrottle(const int &pin, int timeToFull) : Throttle(pin, true)
-{
-  this->m_timeToFull = timeToFull;
-  m_isPressed = false;
-}
-
 int ButtonThrottle::mappingFunction(int time_pressed, m_map_type mapType)
 {
-  if (mapType == LINEAR)
+  if(m_isPressed == true)
+  {
+if (mapType == LINEAR)
   {
-    return std::min((double)time_pressed / (double)m_timeToFull, 1.0) * 100;
+    return std::min((double)time_pressed / (double)m_timeToFull, 1.0) * m_maxValue;
```

Comment:
Is there a reason why the variables aren't declared as doubles either in the calling function or in the parameters for this one? It'd make lines 7, 13 a bit cleaner.